rotkehlchen/tests/utils/dataimport.py

```diff
@@ -1450,3 +1459,181 @@ def assert_binance_import_results(rotki: Rotkehlchen):
         'Skipped 4 rows during processing binance csv file. Check logs for details',
     ]
     assert warnings == expected_warnings
+
+
+def assert_rotki_generic_trades_import_results(rotki: Rotkehlchen):
+    expected_trades = [
+        Trade(
+            timestamp=Timestamp(1659085200),
+            location=Location.EXTERNAL,
+            base_asset=A_USDC,
+            quote_asset=A_ETH,
+            trade_type=TradeType.SELL,
+            amount=AssetAmount(FVal('1.0000')),
+            rate=Price(FVal('1875.64')),
+            fee=None,
+            fee_currency=None,
+            link='092debd43022f98c02d1a7',
+            notes='Trade USDC for ETH',
+        ), Trade(
+            timestamp=Timestamp(1659171600),
+            location=Location.EXTERNAL,
+            base_asset=A_BTC,
+            quote_asset=A_LTC,
+            trade_type=TradeType.BUY,
+            amount=AssetAmount(FVal('4.3241')),
+            rate=Price(FVal('90.85983210379038412617654541')),
+            fee=None,
+            fee_currency=None,
+            link='092debd43022f98c02d1a8',
+            notes='Trade LTC for BTC',
+        ), Trade(
+            timestamp=Timestamp(1659344400),
+            location=Location.EXTERNAL,
+            base_asset=A_UNI,
+            quote_asset=A_DAI,
+            trade_type=TradeType.SELL,
+            amount=AssetAmount(FVal('880.0000')),
+            rate=Price(FVal('0.02272727272727272727272727273')),
+            fee=Fee(FVal('0.1040')),
+            fee_currency=A_USD,
+            link='092debd43022f98c02d1a9',
+            notes='Trade UNI for DAI',
+        ), Trade(
+            timestamp=Timestamp(1659430800),
+            location=Location.EXTERNAL,
+            base_asset=A_BCH,
+            quote_asset=A_ADA,
+            trade_type=TradeType.BUY,
+            amount=AssetAmount(FVal('16.3444')),
+            rate=Price(FVal('280.0127260713149457918308411')),
+            fee=Fee(FVal('5.1345')),
+            fee_currency=A_USD,
+            link='092debd43022f98c02d1a0',
+            notes='Trade ADA for BCH',
+        ),
+    ]
+    with rotki.data.db.conn.read_ctx() as cursor:
+        trades = rotki.data.db.get_trades(cursor, filter_query=TradesFilterQuery.make(), has_premium=True)  # noqa: E501
+        warnings = rotki.msg_aggregator.consume_warnings()
+    assert trades == expected_trades
+    assert len(warnings) == 0
```

Comment:
Add some invalid rows also to verify that they fail as expected and are detected. Check that division by zero are detected for example